forc/Cargo.toml

```diff
@@ -31,7 +31,7 @@ fuel-vm = "0.11"
 futures = "0.3"
 hex = "0.4.3"
 serde = { version = "1.0", features = ["derive"] }
-serde_json = "1.0.73"
+serde_json = { version = "1.0.73", features = ["preserve_order"] }
```

Comment:
Is this necessary for this PR? If not, could you remove and do this in a separate PR?

Reply:
Since I am converting `Vec<u8>` fields to hex strings `serde_json` changes the order the receipt is printed. (data is printed at the beginning if no `preserve_order` is used) if that is not a problem I can omit this one or do that in separate PR.

Reply:
It's not a bad change to please, but please do it in a separate PR if it's not needed to resolve pretty-printing receipts.

Reply:
Ah I may have misunderstood. The issue right now is that `features = ["preserve_order"]` seems to affect the JSON ABI files (see all the changed oracle files). That's what shouldn't be changed as part of this PR is what I meant.

Reply:
Hmm, without changing JSON ABIs I wouldn't be able to use `features = ["preserve_order"]` so I think it makes perfect sense to do that addition in separate PR since it is used to fix ordering not pretty-printing JSON (with hex strings)

Reply:
Is the order in which the receipts show up not important?

Reply:
I am not aware if there would be any technical issues arising from the changed order, I did this to match the current implementation (and other receipt types).

Reply:
The order of _receipts_ is important since receipts are ordered. The order of _fields_ within the receipts isn't important as they all have a key.